### ISSUE TITLE
Perform request issuse 

### ISSUE THREAD

**Issue by Reak45** [thoughtbot/FunctionalJSON-swift#3]
Hi, 
I have a problem. I cannot make this project working. I try to get a user from a web reguest and to do that, I am invoking this function:

 performRequest(request: req) { (user) -> () in
            println("user callback raised!")
        }

But every time i try to build my code I get the following error:
/Users/SM/Downloads/FunctionalJSON-swift-d3fcf771c20813e57cb54472dd8c55ee33e87ae4/FunctionalJSON/ViewController.swift:19:9: Cannot convert the expression's type '(request: @lvalue NSURLRequest, (($T3) -> ()) -> ())' to type 'JSONDecodable'

Could you please help me?
Thanks!

**Comment by alhafoudh** [comment]
I think you need to specify type for the user variable, so instead of this:

```
performRequest(request: req) { (user) -> () in
```

do this:

```
performRequest(request: req) { (user: User) -> () in
```

Creators of this repo started to work on Argo framework using same principles: https://github.com/thoughtbot/Argo

**Comment by Reak45** [comment]
Unfortunatley your solution does not work. Now I get the following error:

/Users/SM/Downloads/FunctionalJSON-swift-d3fcf771c20813e57cb54472dd8c55ee33e87ae4/FunctionalJSON/ViewController.swift:19:9: Type '(request: @lvalue NSURLRequest, (User) -> ())' does not conform to protocol 'JSONDecodable'

And regarding to Argo, as far as I can see there is not a single callback used in the Agro's code

**Comment by alhafoudh** [comment]
Did you make an extension to User with JSONDecodable protocol like this?

```
extension User : JSONDecodable {
...
}
```

**Comment by Reak45** [comment]
Yes, there is an extension to User. I am struggling with this for over 3 days now. I have tried every thing and nothing works. Could be some change in language?

**Comment by alhafoudh** [comment]
I got that working yesterday, but failed on xcode crashing and swift compiler hanging. Can you paste more code?

**Comment by Reak45** [comment]
Yeah sure but I haven't made many changes from default state of project. In fact I added only performRequest method in ViewController class. Here the code:

import UIKit

class ViewController: UIViewController {

```
override func viewDidLoad() {
    super.viewDidLoad()

    var req = NSURLRequest() //I create empty request cause it is not needed for successful build


    performRequest(request: req) { (user : User) -> () in
        println("user callback raised!")
    }
}

override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning()
    // Dispose of any resources that can be recreated.
}
```

}

**Comment by alhafoudh** [comment]
The mapping code would help more..

**Comment by Reak45** [comment]
The mapping code? Could you explain what that is? I am quite new to Swift and I may be missing some things
